### DISCUSSION TITLE
Doubt in knn_graph

### DISCUSSION THREAD

**Discussion by dhorka** [pyg-team/pytorch_geometric#2991] [Q&A] [answered]
Hi, I have a doubt about what is the source and the target when you create a directed graph using the knn_graph method. In flow "source_to_target", the central node is going to be the source or the target? 

Thanks,

**Comment by rusty1s** [comment, marked as answer]
```python
edge_index = knn_graph(x, k=6)
src_indices, dst_indices = edge_index
```
Hope this helps.

**Reply by dhorka** [reply]
My question is more related to the internals of the algorithm. My doubt is how the flow chosen in the algorithm affects its behaviour.  For instance, imagine the following group of nodes.

![Screenshot_20210819_103443](https://user-images.githubusercontent.com/39189267/130036487-949a8fbc-4355-4839-93d0-557d6470f7c9.png)

Let's say that the yellow node is the current central node, and the green ones are the neighbours. My question is in flow "source_to_target", which will be the direction of the edges? The one described in Opt 1 or the one described in Opt2. What is the source, the node or the neighbours?

**Reply by rusty1s** [reply]
The direction will be from green to yellow ones, i.e., Opt1. This ensures that during message passing, each node has exactly `k` neighbors to aggregate messages from.

**Reply by dhorka** [reply]
Thanks! Maybe it could be a good idea to add this clarification to the documentation.

**Reply by rusty1s** [reply]
I agree. Added it to the documentation of `knn_graph`.
